### PR TITLE
[COST-5076] upgrade to python 3.11

### PR DESCRIPTION
**Pull request by maskarb**
## Jira Ticket

[COST-5076](https://issues.redhat.com/browse/COST-5076)

## Description

This change will upgrade to python 3.11.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -33,7 +33,7 @@ LABEL summary="$SUMMARY" \
 # gcc to compile some python packages (e.g. ciso8601)
 # shadow-utils to make useradd available
 # libpq-devel needed for building psycopg2
-RUN INSTALL_PKGS="python39 python39-devel glibc-langpack-en gcc shadow-utils libpq-devel" && \
+RUN INSTALL_PKGS="python3.11 python3.11-devel glibc-langpack-en gcc shadow-utils libpq-devel" && \
```
Comment (samdoran) [inline Dockerfile:36]: Looks like we need the C++ compiler now in order successfully build the wheel for pandas.
```suggestion
RUN INSTALL_PKGS="python3.11 python3.11-devel glibc-langpack-en gcc gcc-c++ shadow-utils libpq-devel" && \
```

**Comment by samdoran** [comment]
We can also update the pre-commit hooks:
```patch
diff --git .pre-commit-config.yaml .pre-commit-config.yaml
index 828eec398..29d46f47c 100644
--- .pre-commit-config.yaml
+++ .pre-commit-config.yaml
@@ -3,16 +3,16 @@ default_language_version:
 
 repos:
   - repo: https://github.com/asottile/reorder_python_imports
-    rev: v3.9.0
+    rev: v3.13.0
     hooks:
       - id: reorder-python-imports
-        args: [--py3-plus, --application-directories, "koku"]
+        args: [--py311-plus, --application-directories, "koku"]
 
   - repo: https://github.com/asottile/pyupgrade
-    rev: v3.3.1
+    rev: v3.15.2
     hooks:
       - id: pyupgrade
-        args: [--py36-plus]
+        args: [--py311-plus]
 
   - repo: https://github.com/ambv/black
     rev: 22.12.0

```

**Review by cgoodfred — DISMISSED** [review]

**Review by lcouzens — APPROVED** [review]

**Pull request merged**